docs/git_configuration.md

```diff
@@ -0,0 +1,13 @@
+## Git Configuration
+
+It is recommended to use this configuration in `.gitconfig` to ensure consistent whitespace rules, improve code readability, prevent unnecessary diffs, and maintain proper file formatting with a final newline.
```

Comment:
also would be helpful to reference https://chromium.googlesource.com/chromium/src/+/main/docs/mac_build_instructions.md#improving-performance-of-git-commands

Reply:
Yes, good idea! 👉 [`aaa57b3` (#29074)](https://github.com/brave/brave-core/pull/29074/commits/aaa57b3d83cd1ce77797d3f5de8a524ec93f3f5e)